IVRv2/app/utils/mongodb.py

```
@@ -1,42 +1,61 @@
 import os
 from pymongo import MongoClient
 from dotenv import load_dotenv
+from app.settings import settings
+from urllib.parse import urlparse
 
 load_dotenv()
 
+
 class MongoDB:
-    def __init__(self, collection_name, db_name="ivr"):
-        connection_string = os.environ.get("MONGO_DB_CONNECTION_STRING")
+    def __init__(self, collection_name):
+        connection_string = settings.mongo_db_connection_string
         if not connection_string or connection_string == "NONE":
             raise ValueError("MONGO_DB_CONNECTION_STRING environment variable not set")
-        
+
         client = MongoClient(connection_string)
-        db = client[db_name]
-        self.collection = db[collection_name]
-    
+
+        # Try to extract database name from connection string
+        db_name = None
+        try:
+            parsed = urlparse(connection_string)
+            # Extract database name from path (e.g., /test or /ivr)
+            path = parsed.path.lstrip("/").split("?")[0]
+            if path:
+                db_name = path
+        except Exception:
+            pass
+
+        # Fallback to environment variable or default
+        if not db_name:
```

No hardcoded fallbacks pls, let there be a startup error thrown if db Name is not specified [AFAIK Mongo uses test db if not specified already]